edenai_apis/apis/sapling/sapling_api.py

```diff
@@ -118,10 +118,10 @@ def text__sentiment_analysis(
         if response.status_code >= HTTPStatus.INTERNAL_SERVER_ERROR:
             raise ProviderException("Internal server error", code=response.status_code)
 
-        response_json = response.json()
-
         if response.status_code >= HTTPStatus.BAD_REQUEST:
```

Comment:
Sometimes, even if status_code is good, when calling `response.json()` can raise a **jsonDecodeError** (I think). There is in sapling a static_mehod avoir that you can use - `_check_error` -